crates/wascc-provider/src/states/pod/starting.rs

```diff
@@ -42,7 +42,7 @@ impl State<PodState> for Starting {
                 Arc::clone(&pod_state.run_context),
             );
             let task_provider = Arc::clone(&provider_state);
-            let task_pod = Arc::clone(&arc_pod);
+            let task_pod = pod_rx.clone();
```

Comment:
Nit: This could be a lot of allocations/clones in the case of a lot of containers. Perhaps an arc is still useful here?

Reply:
Hmmm, `Manifest` requires a `&mut` to use it as a `Stream`. That prevents us from using just an `Arc`, and any other sort of lock would prevent us from streaming concurrently. I think these (`tokio::watch::Receiver`) are fairly lightweight?

Reply:
My worry isn't about the channel so much as the embedded manifest (the inner `T`) which is a pretty large struct

Reply:
Oh, I just realized that it is just the embedded receiver. This is fine then